odbc/src/api/data.rs

```diff
@@ -925,7 +925,7 @@ pub unsafe fn format_bson_data(
                 str_len_or_ind_ptr
             )
         }
-        CDataType::SQL_C_SLONG => {
+        CDataType::SQL_C_SLONG | CDataType::SQL_C_LONG => {
```

Comment:
Like how you added the `CDataType::SQL_C_SLONG | CDataType::SQL_C_LONG => {` for `SQL_C_LONG` you'll want to add     `CDataType::SQL_C_SSHORT | CDataType::SQL_C_SHORT => {` and `CDataType::SQL_C_STINYINT |  CDataType::SQL_C_TINYINT => {` in other parts of this file. 

In the JIRA ticket it mentions `If 2.x, map the datatypes to their signed odbc 3 counterparts`.  I'm not sure where we are checking for the odbc version to map it to the right datatype, do you know @jchemburkar?

Reply:
I think, in fact, this is what I had in mind -- namely that when we call `SQLGetData` with an ODBC type, we should treat it as its corresponding ODBC 3 type.

I wasn't sure, particularly when it came to SQLBindCols, if it would be easier to simply map there (ie bind the cols as the odbc 3 types under the hood), which is why I used that verbiage. But it seems really easy to just handle the odbc 2 c data types in parallel to the 3 ones, so this works for me.